examples/run-blockchain/index.js

```diff
@@ -0,0 +1,172 @@
+
+vm.on('beforeTx', function (tx) {
+  tx._homestead = true
+})
+
+vm.on('beforeBlock', function (block) {
+  block.header.isHomestead = function () {
+    return true
+  }
+})
+
+async.series([
+  // set up pre-state
+  function (next) {
+    setupPreConditions(state, testData, next)
+  },
+
+  // create and add genesis block
+  function (next) {
+    genesisBlock.header = new BlockHeader(
+                            formatBlockHeader(
+                              testData.genesisBlockHeader
+                            )
+                          )
+    blockchain.putGenesis(genesisBlock, next)
+  },
+
+  // add some following blocks
+  function (next) {
+    async.eachSeries(testData.blocks, eachBlock, next)
+
+    function eachBlock (raw, cb) {
+      try {
+        var block = new Block(
+            new Buffer(raw.rlp.slice(2), 'hex'))
+
+        // forces the block into thinking they are homestead
+        block.header.isHomestead = function () {
+          return true
+        }
+        block.uncleHeaders.forEach(function (uncle) {
+          uncle.isHomestead = function () { return true }
+        })
+
+        blockchain.putBlock(block, function (err) {
+          cb(err)
+        })
+      } catch (err) {
+        cb()
+      }
+    }
+  },
+
+  // runBlockchain aka verify all the blocks till date
+  function runBlockchain (next) {
+    vm.runBlockchain(next)
+  },
+
+  // get the blockchain head
+  function getHead (next) {
+    vm.blockchain.getHead(function (err, block) {
+      // make sure the state is set before checking post conditions
+      state.root = block.header.stateRoot
+      next(err)
+    })
+  }
+], function () {
+  console.log('--- Finished processing the BlockChain ---')
+  console.log('New head:', blockchain.meta.rawHead.toString('hex'))
+  console.log('Expected:', testData.lastblockhash)
+})
+
+function formatBlockHeader (data) {
+  var r = {}
+  Object.keys(data)
+        .forEach(function (key) {
+          r[key] = ethUtil.addHexPrefix(data[key])
+        })
+  return r
+}
+
+function setupPreConditions (state, testData, done) {
+  var keysOfPre = Object.keys(testData.pre)
+
+  async.eachSeries(keysOfPre, function (key, callback) {
+    var acctData = testData.pre[key]
+    var account = new Account()
+
+    account.nonce = format(acctData.nonce)
+    account.balance = format(acctData.balance)
+
+    var codeBuf = new Buffer(acctData.code.slice(2), 'hex')
+    var storageTrie = state.copy()
+    storageTrie.root = null
+
+    async.series([
+      function (cb2) {
+        var keys = Object.keys(acctData.storage)
+
+        async.forEachSeries(keys, function (key, cb3) {
+          var val = acctData.storage[key]
+          val = rlp.encode(new Buffer(val.slice(2), 'hex'))
+          key = utils.setLength(new Buffer(key.slice(2), 'hex'), 32)
+
+          storageTrie.put(key, val, cb3)
+        }, cb2)
+      },
+      function (cb2) {
+        account.setCode(state, codeBuf, cb2)
```

Comment:
this stores the contracts code hash in the state trie. We have just save the code to disk (instead of having a unified way to handle values, unfortainly)